File: lib/core/network/api_exception.dart
class ApiException implements Exception {
  final String message;
  final int? statusCode;
  final Map<String, dynamic>? fieldErrors;

  ApiException(this.message, {this.statusCode, this.fieldErrors});

  bool get isAuthError => statusCode == 401;
  bool get isNetworkError => statusCode == null;

  @override
  String toString() => message;
}

File: frontend/src/services/apiException.ts
export class ApiException extends Error {
  statusCode: number | null;
  fieldErrors: Record<string, string> | null;

  constructor(message: string, statusCode: number | null = null, fieldErrors: Record<string, string> | null = null) {
    super(message);
    this.name = "ApiException";
    this.statusCode = statusCode;
    this.fieldErrors = fieldErrors;
  }

  get isAuthError() {
    return this.statusCode === 401;
  }

  get isNetworkError() {
    return this.statusCode === null;
  }
}
